esmvalcore/preprocessor/_multimodel.py

```diff
@@ -53,6 +54,32 @@ def _plev_fix(dataset, pl_idx):
     return statj
 
 
+def _quantile(data, axis, quantile):
+    """Calculate quantile.
+
+    Workaround for calling scipy's mquantiles with arrays of >2 dimensions
+    Similar to iris' _percentiles function, see their discussion:
+    https://github.com/SciTools/iris/pull/625
+    """
+    # Ensure that the target axis is the last dimension.
+    data = np.rollaxis(data, axis, start=data.ndim)
+    shape = data.shape[:-1]
+    # Flatten any leading dimensions.
+    if shape:
+        data = data.reshape([np.prod(shape), data.shape[-1]])
+    # Perform the quantile calculation.
+    result = mstats.mquantiles(data, quantile, axis=-1, alphap=1, betap=1)
```

Comment:
```suggestion
    result = scipy.stats.mstats.mquantiles(data, quantile, axis=-1, alphap=1, betap=1)
```
would be a bit easier to read, because then you can see what function it is without scrolling to the top of the file to look at the import statements